compiler/src/phases/reducing_director.rs

```diff
@@ -384,29 +378,21 @@ impl<R: ReconstructingReducer, O: CombinerOptions> CombineAstAsgDirector<R, O> {
                                 span: span.clone(),
                             }),
                         };
-                        new = ValueExpression::Group(Box::new(group_value));
-                    }
-                    ConstValue::Field(_) => {
-                        new = ValueExpression::Field(tendril.clone(), span.clone());
-                    }
-                    ConstValue::Address(_) => {
-                        new = ValueExpression::Address(tendril.clone(), span.clone());
-                    }
-                    ConstValue::Boolean(_) => {
-                        new = ValueExpression::Boolean(tendril.clone(), span.clone());
-                    }
-                    ConstValue::Char(asg_char) => {
-                        new = match asg_char {
-                            AsgCharValue::Scalar(scalar) => ValueExpression::Char(AstCharValue {
-                                character: Char::Scalar(*scalar),
-                                span: span.clone(),
-                            }),
-                            AsgCharValue::NonScalar(non_scalar) => ValueExpression::Char(AstCharValue {
-                                character: Char::NonScalar(*non_scalar),
-                                span: span.clone(),
-                            }),
-                        }
+                        ValueExpression::Group(Box::new(group_value))
                     }
+                    ConstValue::Field(_) => ValueExpression::Field(tendril.clone(), span.clone()),
+                    ConstValue::Address(_) => ValueExpression::Address(tendril.clone(), span.clone()),
+                    ConstValue::Boolean(_) => ValueExpression::Boolean(tendril.clone(), span.clone()),
+                    ConstValue::Char(asg_char) => match asg_char {
+                        AsgCharValue::Scalar(scalar) => ValueExpression::Char(AstCharValue {
+                            character: Char::Scalar(*scalar),
+                            span: span.clone(),
+                        }),
+                        AsgCharValue::NonScalar(non_scalar) => ValueExpression::Char(AstCharValue {
+                            character: Char::NonScalar(*non_scalar),
+                            span: span.clone(),
+                        }),
+                    },
                     _ => unimplemented!(), // impossible?
```

Comment:
`unreachable`?

Reply:
Yep; also changed the wild-card pattern to an exhaustive mention of the remaining variants so that if new ones are added, we get a compiler error and are forced to consider them here.